### DISCUSSION TITLE
AIO docker install failing Nextcloud not starting on Windows 11

### DISCUSSION THREAD

**Discussion by ccoles146** [nextcloud/all-in-one#2217] [Questions] [answered]
I'm new to docker and have also never used Next Cloud before but I'd like to. It seems pretty straight forward so I appreciate all the work that's gone into making it.
I follow the guide here: https://nextcloud.com/blog/your-guide-to-the-nextcloud-all-in-one-on-windows-10-11/
I created a folder C:\ncdata just so that all the defaults work.
I then go to through the web-based setup, have the port forwarding working, after disabling Cloudflare Proxy. 
Then I disable NextCloud Talk and Enable Collabora, Imaginary and FullTextSearch.
All the containers download and install and start running (according to Docker Desktop). However according to the Containers list, Apache is "starting" and Nextcloud is also starting. Nextcloud actually is constantly restarting and the logs just say something about the install failing and to try again.
I followed the guide to try again and reach exactly the same point again.
The apache container says waiting for Nextcloud to start. 
I did manage to read something in the Nexcloud log last time about the permissions on the data folder being set incorrectly. But what I read about that seemed to indicate that it was more of a warning and that Nextcloud still started.
Here is what I see in the Nextcloud log:

```
2023-03-26 16:27:29 Configuring Redis as session handler...
2023-03-26 16:25:27 + '[' -f /dev-dri-group-was-added ']'
2023-03-26 16:25:27 ++ find /dev -maxdepth 1 -mindepth 1 -name dri
2023-03-26 16:25:27 + '[' -n '' ']'
2023-03-26 16:25:27 + set +x
2023-03-26 16:26:28 + '[' -f /dev-dri-group-was-added ']'
2023-03-26 16:26:28 ++ find /dev -maxdepth 1 -mindepth 1 -name dri
2023-03-26 16:26:28 + '[' -n '' ']'
2023-03-26 16:26:28 + set +x
2023-03-26 16:27:29 + '[' -f /dev-dri-group-was-added ']'
2023-03-26 16:27:29 ++ find /dev -maxdepth 1 -mindepth 1 -name dri
2023-03-26 16:27:29 + '[' -n '' ']'
2023-03-26 16:27:29 + set +x
2023-03-26 16:28:29 + '[' -f /dev-dri-group-was-added ']'
2023-03-26 16:28:29 ++ find /dev -maxdepth 1 -mindepth 1 -name dri
2023-03-26 16:28:29 + '[' -n '' ']'
2023-03-26 16:28:29 + set +x
2023-03-26 16:29:30 + '[' -f /dev-dri-group-was-added ']'
2023-03-26 16:29:30 ++ find /dev -maxdepth 1 -mindepth 1 -name dri
2023-03-26 16:29:30 + '[' -n '' ']'
2023-03-26 16:29:30 + set +x
2023-03-26 16:27:29 Setting php max children...
2023-03-26 16:27:29 The initial Nextcloud installation failed.
2023-03-26 16:27:29 Please reset AIO properly and try again. For further clues what went wrong, check the logs above.
2023-03-26 16:27:29 See https://github.com/nextcloud/all-in-one#how-to-properly-reset-the-instance
2023-03-26 16:28:29 Configuring Redis as session handler...
2023-03-26 16:28:29 Setting php max children...
2023-03-26 16:28:29 The initial Nextcloud installation failed.
2023-03-26 16:28:29 Please reset AIO properly and try again. For further clues what went wrong, check the logs above.
2023-03-26 16:28:29 See https://github.com/nextcloud/all-in-one#how-to-properly-reset-the-instance
2023-03-26 16:29:30 Configuring Redis as session handler...
2023-03-26 16:29:30 Setting php max children...
2023-03-26 16:29:30 The initial Nextcloud installation failed.
2023-03-26 16:29:30 Please reset AIO properly and try again. For further clues what went wrong, check the logs above.
2023-03-26 16:29:30 See https://github.com/nextcloud/all-in-one#how-to-properly-reset-the-instance
```

Thanks.

**Comment by ccoles146** [comment]
Could it be related to this issue? https://github.com/nextcloud/all-in-one/discussions/1873#discussioncomment-4885634
I may try to install the beta version tomorrow then. Unless someone suggests some further guidance.

**Comment by szaimen** [comment]
Hi, can you post the output of `docker run -it --rm --volume nextcloud_aio_nextcloud:/data alpine cat /data/data/install.log` ?

**Reply by ccoles146** [reply]
```
C:\Windows\System32>docker run -it --rm --volume nextcloud_aio_nextcloud:/data alpine cat /data/data/install.log
Unable to find image 'alpine:latest' locally
latest: Pulling from library/alpine
63b65145d645: Already exists
Digest: sha256:ff6bdca1701f3a8a67e328815ff2346b0e4067d32ec36b7992c1fdc001dc8517
Status: Downloaded newer image for alpine:latest
cat: can't open '/data/data/install.log': No such file or directory
```

**Reply by szaimen** [reply]
Hm... And the output of `docker run -it --rm --volume nextcloud_aio_nextcloud:/data alpine ls /data` ?

**Reply by ccoles146** [reply]
That one is blank.

**Reply by szaimen** [reply]
That is strange. Did you already delete all volumes in the meantime?

**Comment by szaimen** [comment]
Anyway, can you reset things properly by following https://github.com/nextcloud/all-in-one#how-to-properly-reset-the-instance and try again?

**Reply by ccoles146** [reply]
Ok, I'll follow that guide to reset things. Then I'll follow the guide above again for the AIO docker install.

**Reply by ccoles146** [reply]
I think I may have previously skipped the update WSL2 Linux kernel step. That may have made a difference with Apache because that is running now (in docker desktop) although it still shows as starting in the container list. aio-nextcloud is still not running.  The command above that gives the same response although alpine does appear to be at least installed: 
''' cat: can't open '/data/data/install.log': No such file or directory '''

**Reply by szaimen** [reply]
Did you make sure to also reset the datadir on the host?
![image](https://user-images.githubusercontent.com/42591237/227798231-c28f8ad7-0d85-48ac-b7b8-091d0f7b9b75.png)

**Reply by ccoles146** [reply]
I used this command 

``` 
docker volume rm nextcloud_aio_nextcloud_datadir 
```
and got the response:

```
Error: No such volume: nextcloud_aio_nextcloud_datadir
```
I did not configure a NEXTCLOUD_DATADIR - I presume this is something to do with the Windows environment variables is it?

In case it's relevant I also got this output when running the docker run command:

```
docker run errors:
Trying to fix docker.sock permissions internally...
Adding internal www-data to group root
WARNING: No blkio throttle.read_bps_device support
WARNING: No blkio throttle.write_bps_device support
WARNING: No blkio throttle.read_iops_device support
WARNING: No blkio throttle.write_iops_device support
```

**Reply by szaimen** [reply]
> I did not configure a NEXTCLOUD_DATADIR - I presume this is something to do with the Windows environment variables is it?

Yes: 
![image](https://user-images.githubusercontent.com/42591237/227798668-57a1a94d-dc36-4086-a899-6722753e96c7.png)

**Reply by szaimen** [reply]
So if you did not change it, you'll still need to clean that location which is located in `C:\ncdata`.

**Reply by ccoles146** [reply]
Thanks for the clarification. So when I manually delete the contents of C:\ncdata and do a reinstall then I get back to the original problem, which looks to me to be permissions related. Here's the log output:

```
Installing with PostgreSQL database
Starting Nextcloud installation...
Nextcloud was successfully installed
Your data directory is readable by other users.
Please change the permissions to 0770 so that the directory cannot be listed by other users.

An unhandled exception has been thrown:
Exception: Environment not properly prepared. in /var/www/html/lib/private/Console/Application.php:166
Stack trace:
#0 /var/www/html/console.php(99): OC\Console\Application->loadCommands(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#1 /var/www/html/occ(11): require_once('/var/www/html/c...')
#2 {main}Your data directory is readable by other users.
Please change the permissions to 0770 so that the directory cannot be listed by other users.

An unhandled exception has been thrown:
Exception: Environment not properly prepared. in /var/www/html/lib/private/Console/Application.php:166
Stack trace:
#0 /var/www/html/console.php(99): OC\Console\Application->loadCommands(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#1 /var/www/html/occ(11): require_once('/var/www/html/c...')
#2 {main}Waiting for appdata to become available...
Waiting for appdata to become available...
Waiting for appdata to become available...
```

**Reply by szaimen** [reply]
I see, so this seemingly wasnt actually fixed even though I tested it and it worked on my side. Will need to come up with a different fix then.

**Comment by szaimen** [comment, marked as answer]
This is now fixed with v4.8.0 Beta. Testing and feedback is welcome! See https://github.com/nextcloud/all-in-one#how-to-switch-the-channel

**Reply by ccoles146** [reply]
Well, it is different but still ...   actually, it is working! It took longer than expected. Let me explain.
The first time the master container started and it suggested that I could use Nextcloud v25 or v26 so I selected v26. It then got stuck on the update. I stopped all containers after 5 minutes and started them again. I think then it took 4 minutes to start up but finally it started.  However, it would appear that it is still running Nexcloud v25. Anyway, thanks for the fix.
